app/models/concerns/orchestration/dhcp.rb

```diff
@@ -93,8 +93,8 @@ def queue_dhcp_update
 
   # do we need to update our dhcp reservations
   def dhcp_update_required?
-    # IP Address / name changed
-    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet))
+    # IP Address / name changed, or 'rebuild' action is triggered.
+    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet) or (old.build? != build? and build?))
```

Comment:
would a better option to query if its valid? AFAIR the record has a valid? and conflicting? methods.
this would blindly delete and recreate the reservation every time?